dcc-portal-server/src/main/java/org/icgc/dcc/portal/server/resource/ui/UISoftwareResource.java

```diff
@@ -54,10 +58,11 @@
   private static final String PUBLIC_KEY_FILE_NAME = "icgc-software.pub";
   private static final String PUBLIC_KEY_PATH = "data/" + PUBLIC_KEY_FILE_NAME;
 
+  private final SoftwareProperties properties;
```

Comment:
Please move `properties` to `SoftwareService` and remove it from the method calls below. I'm pretty sure it will be fully encapsulated and only needed there.